### ISSUE TITLE
incompatible with rails-5.0.0.beta2

### ISSUE THREAD

**Issue by ftiasch** [makandra/active_type#61]
The following code
```ruby
class UserSession < ActiveType::Object
end

UserSession.new
```
generated 
```
ActiveRecord::StatementInvalid: Could not find table 'objects'
```
error.

**Comment by ichord** [comment]
I'm trying to solve this, but why ActiveType::Object need to inherited from ActiveRecord?
We just need `extend ActiveModel::Naming` , `extend ActiveModel::Naming` , `include include ActiveModel::Validations` and `include ActiveSupport::Callbacks`.

**Comment by henning-koch** [comment]
In practice just including `ActiveModel::Model` gives you nowhere near the functionality that ActiveRecord (and hence ActiveType) has.

**Comment by ftiasch** [comment]
@henning-koch but we wants is that models act as a true ActiveRecord, right? just including `ActiveModel::Model` does not give us a `#save`.

**Comment by ichord** [comment]
We can define those active record callbacks by ActiveSupport::Callbacks on our own.

**Comment by ftiasch** [comment]
Yah, I knew this one. But i think `active_type` tends to reuse most things Rails already done for us, minus things we don't need. The new approach sounds like we are mocking a brand new `ActiveRecord`.

Which way should we go?

**Comment by henning-koch** [comment]
We started trying to build on ActiveModel and it didn't work out for what we were trying to do. The idea that ActiveModel is all of AR minus the database is a fairy tale. I don't see us ever going back there.

**Comment by ichord** [comment]
OK.

**Comment by ichord** [comment]
Solved it. but there is problem of type casting. I will work on that later..

**Comment by abhinag739** [comment]
Hi @ichord 
   A very useful gem. Thanks for all the hard work. Can you please help me out :)
 Im using v0.5.1 , on rails 5.0.0.beta4, having the same issue

ActiveRecord::StatementInvalid in VideoUploadsController#new
Could not find table 'objects'
            
class VideoUploadsController < ApplicationController
  def new
    @video_upload = VideoUpload.new
  end

**Comment by lazybios** [comment]
@ichord  the conclusion is what?

**Comment by ichord** [comment]
@lazybios I have tried to fix one of the issues, but didn't have spare time to finish it. Maybe later will keep going on, for we heavily use it on our primary project, unless my leader don't want to upgrade to rails5.0

**Comment by wjp2013** [comment]
`ActiveModel` and [Virtus](https://github.com/solnic/virtus) works on rails 5.0.0.rc1. Maybe you guys should to take a look at Virtus.

* [Creating Form Objects with ActiveModel and Virtus](https://github.com/solnic/virtus)
* [Be nice to others and your future-self: use Data Objects](http://brewhouse.io/2015/07/31/be-nice-to-others-and-your-future-self-use-data-objects.html)
* [Form Objects with Virtus](http://hawkins.io/2014/01/form_objects_with_virtus/)

**Comment by cerdiogenes** [comment]
I'm a bit afraid in using Virtus, due this post from it's maintainer: http://solnic.eu/2016/05/22/my-time-with-rails-is-up.html

@ichord do you know what modification in Rails 5 caused this?

**Comment by cerdiogenes** [comment]
@ichord what do you think is missing in your work? I checkout your changes, run all tests and everything passed. Maybe a gemfiles/Gemfile.5.0 files?

**Comment by cerdiogenes** [comment]
I added gemfiles/Gemfile.5.0.pg with the following:
```
source 'https://rubygems.org'

gem 'activerecord', '~>5.0.0'
gem 'rspec', '~>3.4'
gem 'pg'

gem 'active_type', :path => '..'
```
Now I have tons of warnings and some failed specs to play with, nice!

**Comment by kratob** [comment]
I'll be looking into how to fix this for Rails 5 as well.
The main issue is that Rails 5 already defines `#attribute` for ActiveRecord. Hopfully, we can simply use that and just not run our own "virtual attributes" code on Rails 5.

**Comment by cerdiogenes** [comment]
running the test suite against activerecord-5.0.0 I'm getting the following failure:
```
ActiveRecord::SubclassNotFound:
       Invalid single-table inheritance type: STISpec::Child is not a subclass of STISpec::ExtendedChild
```
These new checks are causing the failure: https://github.com/rails/rails/blob/v5.0.0/activerecord/lib/active_record/inheritance.rb#L188

Any suggestion in how to overcome this?

**Comment by kratob** [comment]
By doing this: https://github.com/makandra/active_type/blob/04533397e4cc14c79e1d7ccd71b4203dec2bb083/lib/active_type/extended_record/inheritance.rb#L26-L63
It's unfortunate, but I don't see another way.

**Comment by kratob** [comment]
I've just released version 0.6.0, which should hopefully fix all the Rails 5 problems.
